shopify/api_access.py

```diff
@@ -0,0 +1,40 @@
+import re
+
+
+class ApiAccess:
+
+    SCOPE_DELIMITER = ","
+    IMPLIED_SCOPE_RE = re.compile(r"\A(?P<unauthenticated>unauthenticated_)?write_(?P<resource>.*)\Z")
+
+    def __init__(self, scopes):
+
+        if type(scopes) == str:
+            scopes = scopes.split(self.SCOPE_DELIMITER)
+
+        self.__store_scopes(scopes)
+
+    def covers(self, api_access):
+        return api_access._compressed_scopes <= self._expanded_scopes
+
+    def __str__(self):
+        return self.SCOPE_DELIMITER.join(self._compressed_scopes)
+
+    def __iter__(self):
+        return iter(self._compressed_scopes)
+
+    def __eq__(self, other):
+        return type(self) == type(other) and self._compressed_scopes == other._compressed_scopes
+
+    def __store_scopes(self, scopes):
```

Comment:
Just out of curiosity what do double leading underscores do? 😅  Is this for name mangling?

Reply:
This is the Python convention of defining "private" methods. Although, methods in Python cannot be truly private. This is just a way to indicate that this method should not be consumed as a public interface as it is more likely to change.

Reply:
ohh okay makes sense!